python/nano/src/bigdl/nano/quantization/neural_compressor/pytorch/utils/dataloader.py

```diff
@@ -0,0 +1,62 @@
+#
+# Copyright 2016 The BigDL Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import torch
+
+
+def _check_data_type(data):
+    if isinstance(data, torch.Tensor):
+        return
+    else:
+        for x in data:
+            if not isinstance(x, torch.Tensor):
+                raise ValueError
+
+
+def _check_loader(model, loader):
+    if loader is None:
+        return
+    sample = next(iter(loader))
+    try:
+        if len(sample) == 2:
+            x, y = sample
+            _check_data_type(x)
+            _check_data_type(y)
+            if isinstance(x, torch.Tensor):
+                model(x)
+            else:
+                model(*x)
+        else:
+            # If sample is not tuple of length 2, then it should be (x1, x2, x3, ...).
+            # Each one must be of torch.Tensor
+            _check_data_type(sample)
+            # check if datalader yields data complied with what model requires
+            # TypeError will be raised if it fails
+            model(*sample)
+    except (ValueError, TypeError):
+        raise ValueError(
+            "Dataloader for quantization in INC should yield data in format below:\n"
```

Comment:
`Dataloader for quantization should yield data in format below:`

Reply:
resolved.